### ISSUE TITLE
Switch CommandEvent to fire before icommand.checkPermission in CommandHandler

### ISSUE THREAD

**Issue by mysticdrew** [MinecraftForge/MinecraftForge#4491]
I want to move away from using a core mod for command permissions.
net.minecraft.command.CommandHandler#executeCommand the CommandEvent is fired after the permission check. Would it be possible to move the event firing to be apart of the permission check. I currently do this with ASM but I know how much core mods are frowned upon. So I'd like to move away from the coremod. 

My mod allows server admins to give users specific commands for moderation with out giving them OP. The commands are most vanilla and all mod commands, they can give them any number of commands. The CommandEvent does not work for this case because the checkPermissions will stop a command before the event is fired if the user is not an OP. 

In short, I'd like the CommandEvent allow command execution that ICommand#checkPermission would not allow under vanilla rules. 

Thanks
-Drew

**Comment by mysticdrew** [comment]
Closing because it is duplicated
